File: src/main/scala/com/github/kazuhito_m/twitterbattler/primitive/domain/model/battle/Battle.scala
package com.github.kazuhito_m.twitterbattler.primitive.domain.model.battle

import com.github.kazuhito_m.twitterbattler.primitive.domain.model.battle.result.{NoResult, Result}

class Battle(
              val turns: Turns,
              val result: Result
            ) {

  def this(turns: Turns) {
    this(turns, NoResult)
  }

  def withResult(result: Result): Battle = new Battle(this.turns, result)

  def hasEnded(): Boolean = result.enable;

}

File: server/src/main/java/com/github/kazuhito_m/twitterbattler/primitive/domain/model/battle/Battle.java
package com.github.kazuhito_m.twitterbattler.primitive.domain.model.battle;

import com.github.kazuhito_m.twitterbattler.primitive.domain.model.battle.result.Result;

public class Battle {
    final Turns turns;
    final Result result;

    public Battle(Turns turns, Result result) {
        this.turns = turns;
        this.result = result;
    }


    public Battle withResult(Result result) {
        return new Battle(this.turns, result);
    }

    public boolean hasEnded() {
        result.enable();
    }

}
